src/alphafold_impact/pipelines/data_collection_oa/nodes.py

```diff
@@ -0,0 +1,120 @@
+import logging
+from typing import List
+import requests
+from requests.adapters import HTTPAdapter, Retry
+import pandas as pd
+
+
+logger = logging.getLogger(__name__)
+
+
+def create_concept_year_filter(concept_ids: List[str], years: List[int]) -> str:
+    """
+    Creates an API query filter string for the OpenAlex API to retrieve works
+    based on lists of concept IDs and years.
+
+    Args:
+        concept_ids: A list of concept IDs (e.g., ['c12345', 'c67890']).
```

Comment:
Arguments and Returns should also include typing hints in docstring. I'll implement it.

Reply:
Thanks. I'll do both typehints and types for the docstring args from now on.